### ISSUE TITLE
Command "python setup.py egg_info" failed with error code 1

### ISSUE THREAD

**Issue by barrierwings-jay** [sunscrapers/djoser#174]
Hi,

I am using Windows OS and Ubuntu at the same time. I didn't have problem to pip install djoser in Ubuntu environment but I got error like below when I tried installing djoser in Window command prompt.

(myvenv) C:\Users\jaemyun\jayground\barrierwings\cango>pip install djoser
Collecting djoser
  Using cached djoser-0.5.1.tar.gz
    Complete output from command python setup.py egg_info:
    See http://johnmacfarlane.net/pandoc/installing.html
    for installation options
    ---------------------------------------------------------------

```
Traceback (most recent call last):
  File "C:\Users\jaemyun\AppData\Local\Temp\pip-build-b9vo40aq\djoser\setup.py", line 8, in <module>
    description = pypandoc.convert('README.md', 'rst')
  File "C:\Users\jaemyun\jayground\barrierwings\cango\myvenv\lib\site-packages\pypandoc\__init__.py", line 69, in convert
    outputfile=outputfile, filters=filters)
  File "C:\Users\jaemyun\jayground\barrierwings\cango\myvenv\lib\site-packages\pypandoc\__init__.py", line 238, in _convert_input
    _ensure_pandoc_path()
  File "C:\Users\jaemyun\jayground\barrierwings\cango\myvenv\lib\site-packages\pypandoc\__init__.py", line 492, in _ensure_pandoc_path
    raise OSError("No pandoc was found: either install pandoc and add it\n"
OSError: No pandoc was found: either install pandoc and add it
to your PATH or or call pypandoc.download_pandoc(...) or
install pypandoc wheels with included pandoc.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Users\jaemyun\AppData\Local\Temp\pip-build-b9vo40aq\djoser\setup.py", line 10, in <module>
    description = open('README.md').read()
UnicodeDecodeError: 'cp949' codec can't decode byte 0xe2 in position 2487: illegal multibyte sequence

----------------------------------------
```

Command "python setup.py egg_info" failed with error code 1 in C:\Users\jaemyun\AppData\Local\Temp\pip-build-b9vo40aq\djoser\

I searched a solution for this problem in internet for quite long, but I couldn't fix this errors.

my pip list

Django (1.10.2)
ez-setup (0.9)
pip (8.1.2)
psycopg2 (2.6.2)
pypandoc (1.2.0)
setuptools (28.6.1)
wheel (0.29.0)

my OS

windows 10 home version 1607

I am kind of new for django. I am sorry if it is not error relating to this djoser package and it caused solely by my window environment.

Thank you!

**Comment by mcgeeco** [comment]
I'm also having decode problems with Fedora:
```
  Using cached djoser-0.5.1.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "/tmp/pip-build-s5ok3kua/djoser/setup.py", line 7, in <module>
        import pypandoc
    ImportError: No module named 'pypandoc'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-s5ok3kua/djoser/setup.py", line 10, in <module>
        description = open('README.md').read()
      File "/root/.ve/em/lib/python3.4/encodings/ascii.py", line 26, in decode
        return codecs.ascii_decode(input, self.errors)[0]
    UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 2487: ordinal not in range(128)
```

**Comment by haxoza** [comment]
Hey @barrierwings-jay & @mcgeeco, thanks for reporting this issue!

It started to be a problem while folks needed to install djoser using setup.py just because the latest version is not in cheese shop yet.

The problem was that `open()` sets local system encoding by default (cp949 and ascii) in your cases. I fixed that in 9b2e81b using `io` module which allows to set `encoding` parameters in `open` function. I also removed non-ascii characters just to get rid of this ambiguity.

Thanks a lot and we're sorry for a long delay!
